src/app/cli/CommandLineInterface.java

```diff
@@ -520,6 +541,295 @@ public void launch(){
+                    }
+                    else if (tokens[2].equals("date")){
+                        Date release_date = new Date(Long.parseLong(tokens[1]));
+                        Game[] game_array = app.search_game_release_date(release_date);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("developer")){
+                        Game[] game_array = app.search_game_developer(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else if (tokens[2].equals("price")){
+                        Game[] game_array = app.search_game_price(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+
+                    }
+                    else if (tokens[2].equals("genre")){
+                        Game[] game_array = app.search_game_genre(tokens[1]);
+                        games_list = new ArrayList<>(Arrays.stream(game_array).toList());
+                    }
+                    else {
+                        System.out.println("\nUsage: ");
+                        System.out.println("SEARCH_GAME { search_val | search_type | sort_val | descend }");
+                        System.out.println("search_val      value being searched");
+                        System.out.println("search_type     type that the being search is, below are acceptable inputs");
+                        System.out.println("                {title},{platform},{date},{developer},{price},{genre}");
+                        System.out.println("sort_val        optional argument to sort the results by, below are acceptable inputs");
+                        System.out.println("                {title},{price},{genre},{developer},{date}");
+                        System.out.println("descend         makes sort into a descend if val=\"D\" otherwise stays ascended\n");
+                        continue;
+                    }
+                    Comparator<Game> default_comparator = new Comparator<Game>() {
+                        @Override
+                        public int compare(Game o1, Game o2) {
+                            int result = o1.title().compareTo(o2.title());
+                            if(result == 0){
+                                return o1.date_release().compareTo(o2.date_release());
+                            } else {
+                                return result;
+                            }
+                        }
+                    };
+                    Comparator<Game> title_comparator = new Comparator<Game>() {
+                        @Override
+                        public int compare(Game o1, Game o2) {
+                            int result;
+                            if((tokens.length>3)&&(tokens[4].equals("D"))) {
```

Comment:
dont do this check in the comparator, you can easily reverse a comparator [like this](https://stackoverflow.com/questions/32995559/reverse-a-comparator-in-java-8)